src/util/eventbus.ts

```diff
@@ -0,0 +1,103 @@
+type Key = symbol | string;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type EventHandler = (...args: any[]) => void;
+
+export type RegisterOptions = {
+  override?: boolean;
+  append?: boolean;
+  insertBefore?: number;
+};
+
+export class EventBus {
```

Comment:
What're the advantages of this EventBus implementation comparing to some classic emitter/pubsub?

- https://github.com/developit/mitt (API: on/off/emit)
- https://www.npmjs.com/package/eventemitter2
- https://www.npmjs.com/package/events

Reply:
It will do better on register event handler, it can provide 3 options to insert event handler append, insertBefore and override, I didn't see that in the above 3 library. And I think use it can provide more flexibility.
On the other hand, if we don't care about the order of event handler, I prefer to use eventemitter2, it seems provide a more powerful event handler.